### PR TITLE
Option to disable vectorization 

### PR DESCRIPTION
**Pull request by newling**
While it will be unusual to not use the default here (bfloats and small integer types **must** currently be vectorized in all situations because AIE does not support scalar operations on these types. And types of size >= 32 bits **must not** be vectorized because AIE does not have vector operations for these), it might be useful for debugging purposes to disable vectorization. If/when peano can emulate scalar bfloat operations, this might be especially useful to switch between vectorized and unvectorized lowering.

### CODE REVIEW

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp [hunk 1]
```diff
@@ -55,6 +48,23 @@ static llvm::cl::opt<std::string> clPathToUkernels(
     "iree-amdaie-path-to-ukernels",
     llvm::cl::desc("Path to microkernels' directory"));
 
+static llvm::cl::opt<bool> clEnableVectorizationInPassPipeline(
+    "iree-amdaie-enable-vectorization-in-pass-pipeline",
```
Comment (yzhang93) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp:52]: I think flag name `iree-amdaie-enable-vectorization` is enough?

Comment (newling) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp:52]: Ok. I actually initially had that, but then thought it was weird that something like 

```
iree-opt --iree-amdaie-enable-vectorization=0 --iree-amdaie-vectorization
```

does vectorize. But on seconds thoughts it's a weird thing to do. I'll change it tomorrow

Comment (yzhang93) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp:52]: Okay, I see. Then maybe `iree-amdaie-enable-vectorization-passes`?

Comment (newling) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp:52]: A good compromise, I'll run with that.

**Review by yzhang93 — APPROVED** [review]
LGTM. Another minor comment... Maybe move the test `disable_vectorization.mlir` to the tests under compiler/.

**Pull request merged**